assets/js/modules/analytics/components/dashboard/LegacyAnalyticsDashboardWidgetTopLevel.js

```diff
@@ -47,174 +46,199 @@ import {
-	}
+function LegacyAnalyticsDashboardWidgetTopLevel( { data, requestDataToState } ) {
+	const [ goals, setGoals ] = useState( false );
+	const [ totalUsers, setTotalUsers ] = useState( false );
+	const [ previousTotalUsers, setPreviousTotalUsers ] = useState( false );
+	const [ overview, setOverview ] = useState( false );
+	const [ extractedAnalytics, setExtractedAnalytics ] = useState( false );
 
-	componentDidMount() {
-		this.processCallbackData();
-	}
+	useEffect( () => {
+		if ( data && ! data.error && 'function' === typeof requestDataToState ) {
+			const {
+				goals: goalsData,
+				overview: overviewData,
+				extractedAnalytics: extractedAnalyticsData,
+				totalUsers: totalUsersData,
+				previousTotalUsers: previousTotalUsersData,
+			} = requestDataToState( { goals, overview, extractedAnalytics, totalUsers, previousTotalUsers }, { data } ) || {};
 
-	/**
-	 * Process callback data received from the API.
-	 */
-	processCallbackData() {
-		const {
-			data,
-			requestDataToState,
-		} = this.props;
+			if ( undefined !== goalsData ) {
+				setGoals( goalsData );
+			}
 
-		if ( data && ! data.error && 'function' === typeof requestDataToState ) {
-			this.setState( requestDataToState );
+			if ( undefined !== overviewData ) {
+				setOverview( overviewData );
+			}
+
+			if ( undefined !== extractedAnalyticsData ) {
+				setExtractedAnalytics( extractedAnalyticsData );
+			}
+
+			if ( undefined !== totalUsersData ) {
+				setTotalUsers( totalUsersData );
+			}
+
+			if ( undefined !== previousTotalUsersData ) {
+				setPreviousTotalUsers( previousTotalUsersData );
+			}
 		}
-	}
+	}, [ data, requestDataToState, goals, overview, extractedAnalytics, totalUsers, previousTotalUsers, setGoals, setOverview, setExtractedAnalytics, setTotalUsers, setPreviousTotalUsers ] );
 
-	render() {
-		const {
-			overview,
-			extractedAnalytics,
-			goals,
-			totalUsers,
-			previousTotalUsers,
-		} = this.state;
+	const { permaLink } = global._googlesitekitLegacyData;
 
-		const { permaLink } = global._googlesitekitLegacyData;
+	const accountID = useSelect( ( select ) => select( STORE_NAME ).getAccountID() );
+	const profileID = useSelect( ( select ) => select( STORE_NAME ).getProfileID() );
+	const internalWebPropertyID = useSelect( ( select ) => select( STORE_NAME ).getInternalWebPropertyID() );
+	const url = useSelect( ( select ) => select( CORE_SITE ).getCurrentEntityURL() );
 
-		const href = getSiteKitAdminURL( 'googlesitekit-module-analytics', {} );
-		const goalURL = 'https://support.google.com/analytics/answer/1032415?hl=en#create_or_edit_goals';
+	const pathIDSegment = `a${ accountID }w${ internalWebPropertyID }p${ profileID }/`;
+	let pathPageSegment = '';
 
-		let goalCompletions = '',
-			goalCompletionsChange = '',
-			averageBounceRate = '',
-			averageBounceRateChange = '';
+	if ( url ) {
+		const parsedURL = new URL( url );
+		pathPageSegment += `_r.drilldown=analytics.pagePath:${ parsedURL.pathname.replace( /\//g, '~2F' ) }`;
```

Comment:
Would be great to introduce a utility function to handle the URL path segment like that.